src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -88,20 +88,14 @@ public static class HostWriter
             }
             catch (Exception ex)
             {
-                FailedToDeleteApphostException failedToDeleteApphostException = null;
                 // Delete the destination file so we don't leave an unmodified apphost
                 try
                 {
                     File.Delete(appHostDestinationFilePath);
                 }
                 catch (Exception failedToDeleteEx) when (failedToDeleteEx is IOException || failedToDeleteEx is UnauthorizedAccessException)
```

Comment:
```suggestion
                catch (Exception failedToDeleteEx)
```

This way, we will not hide the original exception ever in the top-level msbuild error that says the task failed.

Reply:
make sense